### PR TITLE
Add chatterjee facets and innate, slight rework

### PR DESCRIPTION
**Pull request by chrisinajar**
* Change shard from togglable shield to instead refund 90% of remaining (if any) shield cost when shield expires or is recast
* Add facet: Direct Current
* * Allows energy shield to be auto-cast
* * Auto cast only works when CJ is not stunned or silenced and has enough mana to cast
* * Slightly increases defensive numbers
* Add facet: Alternating Current
* * Adds a cooldown to energy shield, no longer allowing it to be up at all times
* * Adds energy shield damage to right clicks when active
* * Slight increases to damage numbers
* Add innate: Battery Powered
* * Chatterjee's maximum mana can never be below her maximum health
* Increase minimum mana required for shield from 5 to 20
* Lowered starting HP
* Slightly increased stat gain

### CODE REVIEW

File: game/scripts/npc/abilities/talents/electrician_talent5_oaa.txt [hunk 1]
```diff
@@ -3,23 +3,12 @@
   //=================================================================================================================
   // Electrician: Electric Shield bonus damage per second
   //=================================================================================================================
-  "special_bonus_electrician_electric_shield_damage"
+  "special_bonus_unique_electrician_electric_shield_damage"
```
Comment (chrisinajar) [inline game/scripts/npc/abilities/talents/electrician_talent5_oaa.txt:6]: renaming this ability *actually* changes how it behaves. if you leave all my other changes but revert the rename then it doesn't actually work lolololololol (starting with `special_bonus_unique_` makes special things happen)

File: game/resource/English/npc/tooltip_electrician.txt [hunk 2]
```diff
@@ -74,7 +74,7 @@
 // talents
 "DOTA_Tooltip_Ability_special_bonus_electrician_energy_absorption_mana"         "+{s:value}% Energy Absorption Max Mana Absorbed"
 "DOTA_Tooltip_Ability_special_bonus_electrician_static_grip_damage"             "+{s:value_tooltip}% Static Grip Damage"
-"DOTA_Tooltip_Ability_special_bonus_electrician_electric_shield_damage"         "+{s:value} Electric Shield Damage"
+"DOTA_Tooltip_Ability_special_bonus_unique_electrician_electric_shield_damage"  "+{s:bonus_aura_damage} Electric Shield Damage"
```
Comment (DarkoniusXNG) [inline game/resource/English/npc/tooltip_electrician.txt:77]: why are you changing the name?

Comment (chrisinajar) [reply game/resource/English/npc/tooltip_electrician.txt:77]: .... im not?

Comment (chrisinajar) [reply game/resource/English/npc/tooltip_electrician.txt:77]: oh the unique part, yeah that actually changes functionality

Comment (chrisinajar) [reply game/resource/English/npc/tooltip_electrician.txt:77]: you can thank valve

Comment (chrisinajar) [reply game/resource/English/npc/tooltip_electrician.txt:77]: https://github.com/OpenAngelArena/oaa/pull/3648#discussion_r1634007788

File: game/scripts/npc/abilities/electrician_electric_shield_oaa.txt [hunk 3]
```diff
@@ -27,47 +27,64 @@
...
         "value"                                           "350"
         "affected_by_aoe_increase"                        "1"
       }
       "aura_damage"
       {
-        "value"                                           "25 35 45 55 110 220"
-        "LinkedSpecialBonus"                              "special_bonus_electrician_electric_shield_damage"
+        "value"                                                           "25 35 45 55 110 220"
+        "special_bonus_unique_electrician_electric_shield_damage"         "+100"
+        "special_bonus_facet_electrician_alternating_current"             "20%"
```
Comment (DarkoniusXNG) [inline game/scripts/npc/abilities/electrician_electric_shield_oaa.txt:70]: is this supposed to be +20% or =20%?

**Comment by chrisinajar** [comment]
conceptually bad, will refactor and reopen later

File: game/resource/English/npc/tooltip_electrician.txt (hunk 2 above)
Comment (DarkoniusXNG) [reply game/resource/English/npc/tooltip_electrician.txt:77]: stupid Valve....

File: game/scripts/npc/abilities/electrician_battery_powered.txt [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+{
+  //=================================================================================================================
+  // Electrician: Battery Powered
+  //=================================================================================================================
+  "electrician_battery_powered"
+  {
+    // General
+    //-------------------------------------------------------------------------------------------------------------
+    "BaseClass"                                           "ability_lua"
+    "AbilityBehavior"                                     "DOTA_ABILITY_BEHAVIOR_PASSIVE | DOTA_ABILITY_BEHAVIOR_NOT_LEARNABLE"
```
Comment (DarkoniusXNG) [inline game/scripts/npc/abilities/electrician_battery_powered.txt:11]: Idk if ```DOTA_ABILITY_BEHAVIOR_INNATE_UI | DOTA_ABILITY_BEHAVIOR_SKIP_FOR_KEYBINDS``` behaviours are relevant.

File: game/scripts/npc/abilities/electrician_battery_powered.txt [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    "ScriptFile"                                          "abilities/electrician/electrician_battery_powered.lua"
+    "RequiredLevel"                                       "-1"
```
Comment (DarkoniusXNG) [inline game/scripts/npc/abilities/electrician_battery_powered.txt:13]: You can take a look how I did custom innate for DK or Underlord. Maybe it's simpler, I am not saying this method is wrong, just using Spawn method avoids using KVs as much as possible.

**Comment by warpdragon** [comment]
looks good. can't seem to test it because "The branch for this pull request no longer exists."

**Pull request closed without merging**